sdk/identity/azure-identity/src/main/java/com/azure/identity/AzurePipelinesCredential.java

```diff
@@ -0,0 +1,100 @@
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.identity.implementation.ClientAssertionCredentialHelper;
+import com.azure.identity.implementation.IdentityClientBase;
+import com.azure.identity.implementation.IdentityClientOptions;
+import com.azure.identity.implementation.models.OidcTokenResponse;
+import com.azure.json.JsonProviders;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.net.URL;
+
+/**
+ * The {@link AzurePipelinesCredential} acquires a token using the Azure Pipelines service connection.
+ *
+ * To construct an instance of this credential, use the {@link AzurePipelinesCredentialBuilder}:
+ * <!-- src_embed com.azure.identity.credential.azurepipelinescredential.construct -->
+ * <pre>
+ * &#47;&#47; serviceConnectionId is retrieved from the portal.
+ * &#47;&#47; systemAccessToken is retrieved from the pipeline environment as shown.
+ * &#47;&#47; You may choose another name for this variable.
+ *
+ * String systemAccessToken = System.getenv&#40;&quot;SYSTEM_ACCESSTOKEN&quot;&#41;;
+ * AzurePipelinesCredential credential = new AzurePipelinesCredentialBuilder&#40;&#41;
+ *     .clientId&#40;clientId&#41;
+ *     .tenantId&#40;tenantId&#41;
+ *     .serviceConnectionId&#40;serviceConnectionId&#41;
+ *     .systemAccessToken&#40;systemAccessToken&#41;
+ *     .build&#40;&#41;;
+ * </pre>
+ * <!-- end com.azure.identity.credential.azurepipelinescredential.construct -->
+ *
+ */
+@Immutable
+public class AzurePipelinesCredential implements TokenCredential {
+    private static final ClientLogger LOGGER = new ClientLogger(AzurePipelinesCredential.class);
+    private final ClientAssertionCredentialHelper clientAssertionCredentialHelper;
+
+    /**
+     * Creates an instance of {@link AzurePipelinesCredential}.
+     *
+     * @param clientId the client id of the service principal
+     * @param tenantId the tenant id of the service principal
+     * @param requestUrl the request url to get the client assertion token
+     * @param systemAccessToken the system access token
+     * @param identityClientOptions the options for configuring the identity client
+     */
+    AzurePipelinesCredential(String clientId, String tenantId, String requestUrl, String systemAccessToken, IdentityClientOptions identityClientOptions) {
+        clientAssertionCredentialHelper = new ClientAssertionCredentialHelper(clientId, tenantId, identityClientOptions, () -> {
+            HttpClient client = identityClientOptions.getHttpClient();
+            if (client == null) {
+                HttpClient.createDefault();
+            }
+            HttpPipeline pipeline = IdentityClientBase.setupPipeline(client, identityClientOptions);
+            try {
+                URL url = new URL(requestUrl);
```

Comment:
This is all impl logic, it can be moved to IdentityClient and accessed directly there with the pipeline instance in the client.